src/services/cronJobs/importLostDonationsJob.ts

```diff
@@ -0,0 +1,279 @@
+          transaction,
+          receipt,
+          tokenInDB,
+          transactionData,
+        );
+
+        if (!donationParams) continue; // transaction not mined yet
+      } else if (
+        !tokenInDB &&
+        transaction.value &&
+        transaction.value.gt(0) &&
+        (!transaction.data || transaction.data === '0x')
+      ) {
+        // it's an eth transfer native token
+        const nativeTokenSymbol = getNetworkNativeToken(networkId);
+        const nativeToken = await Token.createQueryBuilder('token')
+          .where(`token.symbol = symbol`, { symbol: nativeTokenSymbol })
+          .andWhere(`token."networkId" = :networkId`, { networkId })
+          .getOne();
+
+        tokenInDB = nativeToken;
+
+        donationParams = await getDonationDetailForNormalTransfer(
+          tx,
+          networkId,
+          transaction,
+          receipt,
+          nativeToken!,
+        );
+      } else {
+        continue; // Not a transaction recognized by our logic
+      }
+
+      const project = await Project.createQueryBuilder('project')
+        .leftJoinAndSelect('project.addresses', 'addresses')
+        .leftJoinAndSelect('project.adminUser', 'adminUser')
+        .where(`lower(addresses.address) = lower(:address)`, {
+          address: donationParams.to,
+        })
+        .getOne();
+
+      if (!project) continue; // project doesn't exist on giveth, skip donation
+
+      const donationDate = millisecondTimestampToDate(
+        donationParams.timestamp * 1000,
+      );
+      const donationDateDbFormat = moment(donationDate).format(
+        'YYYY-MM-DD HH:mm:ss',
+      );
+      const donationDateCoingeckoFormat =
+        moment(donationDate).format('DD-MM-YYYY');
+
+      const coingeckoAdapter = new CoingeckoPriceAdapter();
+      const ethereumPriceAtDate = await coingeckoAdapter.getTokenPriceAtDate({
+        symbol: tokenInDB!.coingeckoId,
+        date: donationDateCoingeckoFormat,
+      });
+
+      const { givbackFactor, projectRank, powerRound, bottomRankInRound } =
+        await calculateGivbackFactor(project.id as number);
+
+      const dbDonation = Donation.create({
+        fromWalletAddress: donationParams.from.toLowerCase(),
+        toWalletAddress: donationParams.to.toLowerCase(),
+        transactionId: tx.toLowerCase(),
+        projectId: project.id,
+        currency: donationParams.currency,
+        tokenAddress: tokenInDB?.address,
+        amount: donationParams.amount,
+        valueUsd: toFixNumber(ethereumPriceAtDate * donationParams.amount, 4),
+        transactionNetworkId: networkId,
+        createdAt: donationDateDbFormat,
+        givbackFactor,
+        projectRank,
+        powerRound,
+        bottomRankInRound,
+        status: 'verified',
+        anonymous: false,
+        segmentNotified: true,
+        isTokenEligibleForGivback: tokenInDB?.isGivbackEligible,
+        isProjectVerified: project.verified,
+        qfRoundId: qfRound?.id,
```

Comment:
@CarlosQ96  Should not add qfRound to all missed donations, we should check if project is connected to qfRound, then add `qfRoundId` to donation ( and its nice to have a test case for that scenario as well)

Reply:
I did it myself just please take a look yourself too